src/hotspot/share/opto/node.cpp

```diff
@@ -68,30 +68,17 @@ extern int nodes_created;
 // Set a breakpoint here to identify where a particular node index is built.
 void Node::verify_construction() {
   _debug_orig = nullptr;
-  int old_debug_idx = Compile::debug_idx();
-  int new_debug_idx = old_debug_idx + 1;
-  if (new_debug_idx > 0) {
-    // Arrange that the lowest five decimal digits of _debug_idx
-    // will repeat those of _idx. In case this is somehow pathological,
-    // we continue to assign negative numbers (!) consecutively.
-    const int mod = 100000;
-    int bump = (int)(_idx - new_debug_idx) % mod;
-    if (bump < 0) {
-      bump += mod;
-    }
-    assert(bump >= 0 && bump < mod, "");
-    new_debug_idx += bump;
-  }
-  Compile::set_debug_idx(new_debug_idx);
-  set_debug_idx(new_debug_idx);
+  // The decimal digits of _debug_idx are <compile_id> followed by 5 digits of <_idx>
   Compile* C = Compile::current();
   assert(C->unique() < (INT_MAX - 1), "Node limit exceeded INT_MAX");
+  uintx new_debug_idx = (uintx)C->compile_id() * 100000 + _idx;
```

Comment:
Should we assert that _idx < 100000? We can use bigger multiplier since debug_idx is 64 bit value now.

Reply:
I can make the multiplier 10000000000 so we don't have to assert or artificially restrict _idx further than INT_MAX.  I also need to change the type from uintx to uint64_t so I don't break 32-bit ports.